koocook_core/models/review.py

```diff
@@ -185,7 +197,10 @@ def add_rating(self, rating: Rating):
         self.check_rating(rating)
         total_value = self.rating_value * self.rating_count
         total_value += rating.rating_value
-        self.rating_count += 1
+        if not update:
+            self.rating_count += 1
+        else:
+            total_value -= rating.old_rating_value
         self.rating_value = total_value / self.rating_count
         self.save()
```

Comment:
Could you explain to me why it is done this way. This looks a but gimmicky.

Reply:
This is just like subtracting a sample from the summation of the average.

Reply:
ooh I get it now. There's a new 'update' function right. Kk

Reply:
It's an inverse approach to the `add_recipe` function.